common/src/main/java/com/viaversion/viaversion/rewriter/BlockRewriter.java

```diff
@@ -67,15 +67,14 @@ public static <C extends ClientboundPacketType> BlockRewriter<C> for1_20_2(final
     }
 
     public void registerBlockEvent(C packetType) {
-        if (protocol.getMappingData().getBlockMappings() == null) {
-            return;
-        }
-
         protocol.registerClientbound(packetType, wrapper -> {
             wrapper.passthrough(positionType); // Location
             wrapper.passthrough(Types.UNSIGNED_BYTE); // Action id
             wrapper.passthrough(Types.UNSIGNED_BYTE); // Action param
             final int blockId = wrapper.passthrough(Types.VAR_INT);
+            if (protocol.getMappingData().getBlockMappings() == null) {
```

Comment:
You can move the check up before all `passthrough` calls I believe.

Reply:
didn't know thx